circuit_breaker_test.go

```diff
@@ -0,0 +1,242 @@
+package aibridge
```

Comment:
These tests seem to be testing the library, not the _observed behaviour_.

What I'd expect to see from a test is a mock upstream HTTP server returning errors which results in a response indicating a circuit-breaker has tripped.